src/target/target_flash.c

```diff
@@ -0,0 +1,271 @@
+/*
```

Comment:
Just a note: If you wish to license this differently under, eg, BSD-3-Clause, as this is considered a fresh contribution you would be welcome to.